### PR TITLE
Adding Edge Impulse + micro-ROS demo

### PR DESCRIPTION
**Pull request by avielbr**
Hi @ralph-lange ,

Thanks for the invitation to add this demo page. I hope I formatted everything correctly. I will be in touch when I add more examples / additional board support to coordinate with you whether to update this demo page.

### CODE REVIEW

File: _docs/tutorials/demos/edgeimpulse_demo/index.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```
+'dog': 0.75,
+'cat': 0.25
+```
+
+It contains all labels and their probabilities (together summing to 1). The size of the result messages depends on how many labels (classes) your image model is trained on (the example above has two). The code in this example will automatically allocate the necessary memory to support an arbitrary number of labels depending on your model.
+
+
+
+By running neural networks on MCUs and publishing their inferences using micro-ROS it becomes straightforward to experiment with a "distributed" approach to AI in robotics, where the central computer is concerned only with the results of the neural networks and not with sensor data ingest nor calculation.
```
Comment (ralph-lange) [inline _docs/tutorials/demos/edgeimpulse_demo/index.md:34]:
```suggestion
By running neural networks on MCUs and publishing their inferences using micro-ROS, it becomes straightforward to experiment with a "distributed" approach to AI in robotics, where the central computer is concerned only with the results of the neural networks and not with sensor data ingest nor calculation.
```

File: _docs/tutorials/demos/edgeimpulse_demo/index.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+![Traditional vs distributed approach](traditional_vs_distributed.png)
+
+---
+
+
+To run this demo yourself check out the video walk through and instructions given [here](https://github.com/avielbr/micro_ros_ei).
```
Comment (ralph-lange) [inline _docs/tutorials/demos/edgeimpulse_demo/index.md:41]:
```suggestion
To run this demo yourself, check out the video walk through and instructions given [here](https://github.com/avielbr/micro_ros_ei).
```

**Review by ralph-lange — CHANGES_REQUESTED** [review]
Thank you very much for contributing this demo/tutorial. The description is very well done and concise. With two somewhat longer sentences, readability can be increased by a comma, I think. But I am not a native speaker.
Also, you may omit the `_demo' in the folder name since the parent folder is already named `demos`. (I know, all other folders have this unnecessary `_demo`.)

File: _docs/tutorials/demos/edgeimpulse_demo/index.md (hunk 1 above)
Comment (avielbr) [reply _docs/tutorials/demos/edgeimpulse_demo/index.md:34]: Thanks for the suggestion @ralph-lange, I agree this improves readability.

**Comment by avielbr** [comment]
I changed the folder name as requested :) Let me know if there is anything else. Thanks.

**Review by ralph-lange — APPROVED** [review]
Perfect. Thanks again very much!

**Comment by avielbr** [comment]
I realized I hadn't changed the name of the folder within the header as well, so I did that now. I don't know if that is the reason why the build failed..?

**Comment by ralph-lange** [comment]
The build failed because of few broken links in other pages of the website - not in your page.

**Comment by avielbr** [comment]
@ralph-lange sorry to be a nuisance - but shall I also update `overview/index.md` so that the table includes my addition? Also, is there anything I need to do so that the demo appears in the "Demos" dropdown menu on the site?

**Pull request merged**